pkg/controller/direct/logging/utils.go

```diff
@@ -76,6 +76,21 @@ func compareMetricDescriptors(kccObj *krm.LogmetricMetricDescriptor, apiObj *api
 	return reflect.DeepEqual(kccObj, convertAPItoKRM_MetricDescriptor(apiObj))
 }
 
+func validateImmutableFieldsUpdated(kccObj *krm.LogmetricMetricDescriptor, apiObj *api.MetricDescriptor) error {
+	actualMetricDescriptor := convertAPItoKRM_MetricDescriptor(apiObj)
+	modified := []string{}
+	if !reflect.DeepEqual(kccObj.MetricKind, actualMetricDescriptor.MetricKind) {
```

Comment:
I have a helper that might be useful - see #1770

Reply:
Is that PR still WIP or ready? it looks good to me and we can get it merged.

Reply:
Let's go with what you've got for 1.118 and then we can look more in 1.119